### PR TITLE
feat(plugins): added new menu state flags

### PR DESCRIPTION
**Pull request by Skaiir**
Closes [#2757](https://github.com/camunda/camunda-modeler/issues/2757)

### CODE REVIEW

File: client/src/app/tabs/bpmn/BpmnEditor.js [hunk 1]
```diff
@@ -430,6 +430,7 @@ export class BpmnEditor extends CachedComponent {
     // ensure backwards compatibility
     // https://github.com/camunda/camunda-modeler/commit/78357e3ed9e6e0255ac8225fbdf451a90457e8bf#diff-bd5be70c4e5eadf1a316c16085a72f0fL17
     newState.bpmn = true;
+    newState.bpmnC7 = true;
```
Comment (nikku) [inline client/src/app/tabs/bpmn/BpmnEditor.js:433]: Given our clear separation between _platform_ and _cloud_, why not add a more generic `platform` attribute that is either `platform` or `cloud` (or whatever names we come up with). Works great with `generic` (maybe never introduced).

**Review by nikku — CHANGES_REQUESTED** [review]

File: client/src/app/tabs/bpmn/BpmnEditor.js (hunk 1 above)
Comment (Skaiir) [reply client/src/app/tabs/bpmn/BpmnEditor.js:433]: So a plugin dev that would like to make a menu plugin for bpmn c8 would go something like:

`enabled: ( ) => menuState.bpmn && menuState.c8`

No overlap in logic, but two states to check as tradeoff for the plugin developers. Honestly I'm happy to make the change, and also add it to the plugin docs you wrote to give a bit of extra visibility on this common use case, what are your thoughts on that?

Comment (Skaiir) [reply client/src/app/tabs/bpmn/BpmnEditor.js]: I made the change, I think I like this more anyways as well.

Comment (pinussilvestrus) [reply client/src/app/tabs/bpmn/BpmnEditor.js]: I'm not super sure about using `c7` and `c8` at this stage. We don't mention it somewhere in the UI or the code so far.

What about this (and I think that Nico mentioned this as well)?

```js
// bpmn/BpmnEditor.js
newState.platform = 'platform'; // or c7, if we decide to do so

// cloud-bpmn/BpmnEditor.js
newState.platform = 'cloud'; // or c8, ...

// someday in the future, not supported yet
newState.platform = 'generic';
```

Comment (Skaiir) [reply client/src/app/tabs/bpmn/BpmnEditor.js]: @pinussilvestrus

Thanks for the clarification this is actually much nicer and I've changed my first way of doing this.

Are we sure just committing to c7/c8 isn't right at this stage, I think personally that we are going to have to do it at one point, so maybe let's just start doing that. During some chats with @MaxTru we had come to the conclusion that anything new should adopt the new nomenclature. Are we going back on that then?

Comment (MaxTru) [reply client/src/app/tabs/bpmn/BpmnEditor.js]: c7/c8 naming in the sourceCode is indeed super unclear at the moment.

@Skaiir for now: let`s please use the existing `platform` and `cloud` pattern. This way we at least have it consistent in all our sources. And then we can change in a batch in the future. I hope this simplifies the discussions.

Sorry, I stated this differently last week :-)

Comment (Skaiir) [reply client/src/app/tabs/bpmn/BpmnEditor.js]: Alright, I've put this through.

**Review by MaxTru — APPROVED** [review]

**Comment by MaxTru** [comment]
Would it make sense to have a test here?

**Comment by Skaiir** [comment]
I would say no personally. There's a pretty much infinite amount of ways menu plugins can be conditionally enabled and disabled and here all we're doing is exposing some data for people to do so. The newState.bpmn stuff wasn't getting tested either, and I think it makes sense, we'd be testing plugin availability based on our expectations of how people will most likely use them which I think is a bit flawed.

But I'm open to opinions on this one.

**Pull request merged**